sample-apps/delivery-customizations-sample-app/extensions/move-to-last-by-title/src/main.rs

```diff
@@ -0,0 +1,132 @@
+        .iter()
+        .filter_map(|delivery_option| {
+            if &delivery_option_name == &delivery_option.title.as_str() {
+                Some(Operation {
+                    r#move: Some(MoveOperation {
+                        delivery_option_handle: delivery_option.handle.clone(),
+                        index: delivery_options.len() as i32 - 1,
+                    }),
+                })
+            } else {
+                None
+            }
+        })
+        .collect();
+
+    Ok(FunctionResult { operations })
+}
+
+fn delivery_options(cart: &Cart) -> Vec<&CartDeliveryOption> {
+    cart.delivery_groups
+        .iter()
+        .flat_map(|delivery_group| &delivery_group.delivery_options)
+        .collect()
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    fn input(configuration: Option<Configuration>) -> Input {
+        let input = r#"
+        {
+            "cart": {
+                "deliveryGroups": [
+                    {
+                        "deliveryOptions": [
+                            { "handle": "method-a", "title": "Method A" }
+                        ]
+                    },
+                    {
+                        "deliveryOptions": [
+                            { "handle": "method-b", "title": "Method B" },
+                            { "handle": "method-c", "title": "Method C" }
+                        ]
+                    }
+                ]
+            },
+            "deliveryCustomization": {
+                "metafield": {
+                  "value": "Method A"
+                }
+            },
+            "localization": {
+                "country": { "isoCode": "CA" },
+                "language": { "isoCode": "EN" }
+            },
+            "presentmentCurrencyRate": "2.0"
+        }
+        "#;
+        let default_input: Input = serde_json::from_str(input).unwrap();
+        let value = serde_json::to_string(&configuration.unwrap_or_default()).unwrap();
+
+        
+
+        Input {
+            delivery_customization: DeliveryCustomization {
+                metafield: Some(Metafield { value }),
+            },
+            ..default_input
+        }
+    }
+
+    #[test]
+    fn test_with_no_configuration() {
+        let input = input(None);
+        let operations = function(input).unwrap().operations;
+
+        assert!(operations.is_empty());
+    }
+
+    #[test]
+    fn test_move_operations_with_configuration() {
+        let input = input(Some(Configuration {
+            delivery_option_name: "Method A".to_string(),
+        }));
+        let operations = function(input).unwrap().operations;
+
+        assert_eq!(operations.len(), 1);
+        assert_eq!(
+            operations[0]
+                .r#move
+                .as_ref()
+                .unwrap()
+                .delivery_option_handle,
+            "method-a"
+        );
+        assert_eq!(operations[0].r#move.as_ref().unwrap().index, 2);
+    }
+}
```

Comment:
Have something wrong with this tests for each extensions. Not sure if Im just missing something in Rust, but I cant tell why its not working if anyone has a chance to look (no rush) @dphm @jianghong . 

There don't seem to ever be any operations returned in the test, but function-runner is passing, so I think it just with the test, but Im not 100% sure.

Reply:
Do you mind if I add a couple commits to fix the function tests and update to use the `shopify_function` crate?

Reply:
TY!

Reply:
I pushed another commit to use `shopify_function`!

Hope this is the metafield namespace/key you're using in core:

```graphql
  deliveryCustomization {
    metafield(namespace: "delivery-customizations-sample-app", key: "function-configuration") {
      value
    }
  }
```